src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -283,11 +287,67 @@ TransactionManager::get_next_dirty_extents(
   return cache->get_next_dirty_extents(seq, max_bytes);
 }
 
-TransactionManager::rewrite_extent_ret TransactionManager::rewrite_extent(
+TransactionManager::rewrite_extent_ret TransactionManager::rewrite_extents(
   Transaction &t,
-  CachedExtentRef extent)
+  std::vector<CachedExtentRef>& extents)
 {
   LOG_PREFIX(TransactionManager::rewrite_extent);
+  std::vector<LogicalCachedExtentRef> logical_extents;
+  return seastar::do_with(
+    std::vector<LogicalCachedExtentRef>(),
+    [this, &extents, &t, FNAME](auto& logical_extents) mutable {
+    return trans_intr::do_for_each(extents,
+      [this, &logical_extents, &t, FNAME](CachedExtentRef& extent) mutable
+      -> rewrite_extent_ret {
+      if (extent->is_logical()) {
+	logical_extents.emplace_back(extent->cast<LogicalCachedExtent>());
+	return seastar::now();
+      } else {
+	DEBUGT("cleaning {}", t, *extent);
+	return rewrite_physical_extent(t, extent);
+      }
+    }).si_then([this, &logical_extents, &t, FNAME]() mutable {
+      return trans_intr::do_for_each(logical_extents,
+	[this, &t, FNAME](LogicalCachedExtentRef& extent) {
+	auto updated = cache->update_extent_from_transaction(t, extent);
+	if (!updated) {
+	  DEBUGT("{} is already retired, skipping", t, *extent);
+	  return rewrite_extent_iertr::now();
+	}
+	assert(updated->is_logical());
+	extent = updated->cast<LogicalCachedExtent>();
+	DEBUGT("cleaning {}", t, *extent);
+	return lba_manager->rewrite_logical_extent_p1(t, extent);
+      }).si_then([this, &t]() mutable {
+	return trans_intr::parallel_for_each(
+	  t.get_rewrite_block_list(),
+	  [](auto& p) {
+	  auto& writer = p.first;
+	  auto& lextents = p.second;
+	  return writer->write(lextents);
+	}).si_then([this, &t]() mutable {
+	  return trans_intr::do_for_each(t.get_rewrite_block_list(),
+					 [this, &t](auto& p) mutable {
+	    return trans_intr::do_for_each(p.second,
+	      [this, &t](LogicalCachedExtentRef& nlextent) mutable {
+	      auto lextent = nlextent->get_old_extent()->cast<LogicalCachedExtent>();
+	      return lba_manager->rewrite_logical_extent_p2(t, lextent, nlextent);
+	    });
+	  });
+	});
+      });
+    });
+  }).handle_error_interruptible(
+    crimson::ct_error::input_output_error::pass_further{},
+    crimson::ct_error::eagain::pass_further{},
+    crimson::ct_error::assert_all{"Invalid error when rewriting extent"});
+}
```

Comment:
Oh, for one thing, rewriting lba extents are more complex than rewriting logical extents, as we have to figure out a way to resolve all relative address when rewriting them; for another, lba extents tends to modified far more frequently than logical extents, because any write operation issued to seastore will lead to modification of the lba tree, it may not be necessary to rewrite them to other segments, so only logical extents are rewritten to other segments for now. If in later performance testing, it turns out to the other way around, we can implement lba extents rewriting then.